src/server/api/boards/boards.js

```diff
@@ -160,10 +175,43 @@ function createNewColumn(request, username, boardId, callback) {
   });
 }
 
+function deleteColumnId(username, boardId, columnsId, callback) {
+  MongoClient.connect(url, function(err, database) {
+    if (err) {
+      return callback('error');
+    }
+    console.log('Connection established to ' + url);
+    let query = columnQuery(username, boardId, columnsId);
+
+    if (!query) {
+      return callback('notFound');
+    }
+    database.collection('boards').findOne(query, function(err, result) {
+      console.log(result);
+      if (result === null) {
+        return callback('notFound');
+      }
+      let newColumns = result.columns.filter(function(e) {
+        if (e._id.toString() !== new mongodb.ObjectId(columnsId).toString()) {
+          return e;
```

Comment:
Filter should return `true` if we want to keep the element `false` otherwise.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter